CMakeLists.txt

```diff
@@ -123,6 +133,70 @@ if(USE_GPU)
     ADD_DEFINITIONS(-DUSE_GPU)
 endif(USE_GPU)
 
+if(USE_CUDA)
+    find_package(CUDA REQUIRED)
+    include_directories(${CUDA_INCLUDE_DIRS})
+    LIST(APPEND CMAKE_CUDA_FLAGS -g -Xcompiler=-fopenmp -Xcompiler=-fPIC -Xcompiler=-Wall -lineinfo)
+    CUDA_SELECT_NVCC_ARCH_FLAGS(CUDA_ARCH_FLAGS 7.0)
+
+    LIST(APPEND CMAKE_CUDA_FLAGS ${CUDA_ARCH_FLAGS})
+    if(CMAKE_BUILD_TYPE MATCHES Release)
+      LIST(APPEND CMAKE_CUDA_FLAGS -03)
+    endif()
+
+    message(STATUS "CMAKE_CUDA_FLAGS: ${CMAKE_CUDA_FLAGS}")
+    string(REPLACE ";" " " CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS}")
+    message(STATUS "CMAKE_CUDA_FLAGS: ${CMAKE_CUDA_FLAGS}")
+    set(CMAKE_CUDA_FLAGS_DEBUG -G)
+    set(CMAKE_CUDA_FLAGS_RELEASE -lineinfo)
+
+    ADD_DEFINITIONS(-DUSE_CUDA)
+    if (NOT DEFINED CMAKE_CUDA_STANDARD)
+      set(CMAKE_CUDA_STANDARD 11)
+      set(CMAKE_CUDA_STANDARD_REQUIRED ON)
+    endif()
+
+    set(BASE_DEFINES
+     -DPOWER_FEATURE_WORKGROUPS=12
+     -DUSE_CONSTANT_BUF=0
+    )
```

Comment:
Could you please elaborate on these defs?

Reply:
I find references to them here within LightGBM GPU for Windows

https://lightgbm.readthedocs.io/en/latest/GPU-Windows.html

Reply:
Thank you! Have never seen them before.
https://github.com/microsoft/LightGBM/blob/bca2da97284403b1b724a0e2966a76ecbb733fc9/src/treelearner/gpu_tree_learner.cpp#L580-L584